app/src/test/java/com/hedvig/app/feature/embark/passages/ExternalInsurerViewModelTest.kt

```diff
@@ -0,0 +1,93 @@
+package com.hedvig.app.feature.embark.passages
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.hedvig.app.feature.embark.passages.externalinsurer.ExternalInsurerViewModel
+import com.hedvig.app.feature.embark.passages.externalinsurer.GetInsuranceProvidersUseCase
+import com.hedvig.app.feature.embark.passages.externalinsurer.InsuranceProvider
+import com.hedvig.app.feature.embark.passages.externalinsurer.InsuranceProvidersResult
+import com.hedvig.app.util.coroutines.MainCoroutineRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Rule
+import org.junit.Test
+
+class ExternalInsurerViewModelTest {
+
+    private val insuranceProviders = listOf(
+        InsuranceProvider("1", "Test1"),
+        InsuranceProvider("2", "Test2"),
+        InsuranceProvider("3", "Test3")
+    )
+
+    private var insuranceProvidersResult: InsuranceProvidersResult =
+        InsuranceProvidersResult.Success(insuranceProviders)
+
+    @ExperimentalCoroutinesApi
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private val getInsuranceProvidersUseCase = object : GetInsuranceProvidersUseCase {
```

Comment:
Implementing the interface here, in order to test different results (and not depending on apollo and network). I think this works really well! The test does not need to know how the result is created, since we are testing the view model. We also have the possibility to change the implementation of the use case to use a database, another networking solution etc. without affecting the view model test.

Reply:
I see! How does this differ from ```kotlin
val getInsuranceProvidersUseCase = mockk<GetInsuranceProvidersUseCase()

every { getInsuranceProvidersUseCase.getInsuranceProviders() } returns insuranceProvidersResult
```

though? 😄

Reply:
Good point, mocking is much cleaner. Fixed!